### PR TITLE
fby4: sd: Support to get slot ID by IPMI command

### PR DESCRIPTION
**Pull request by BonnieLo**
# Description:
    1. Create the CMD code "CMD_OEM_GET_CHASSIS_POSITION(0x7E)".
    2. Create weak function "OEM_GET_CHASSIS_POSITION" in common code.
    3. Create function "OEM_GET_CHASSIS_POSITION" in yv4-sd.
    4. Add "CMD_OEM_GET_CHASSIS_POSITION(0x7E) into pal_request_msg_to_BIC_from_HOST().

# Motivation:
    This change aims to support the host in retrieving slot ID through IPMI commands.

# Test Plan:
    Send IPMI command(netfn:0x30, cmd:0x7E) to get the slot ID in Host console: Pass

# Test log:
    Check Slot ID in Host console:
        /* In slot4's Host */
        [root@20231109-643fbk1rc21 ~]# ipmitool raw 0x30 0x7e
        84

    	/*   msg->data[0] format:
         *
         *   Slot Present (Bit 7):
         *       0: Slot not present
         *       1: Slot present
         *
         *   Reserved (Bits [6:4])
         *
         *   Slot ID (Bits [3:0]):
         *       0001: Slot 1
         *       0010: Slot 2
         *       ...
         *       1000: Slot 8
         */

### CODE REVIEW

File: meta-facebook/yv4-sd/src/ipmi/plat_ipmi.c [hunk 1]
```diff
@@ -112,3 +119,30 @@ void APP_COLD_RESET(ipmi_msg *msg)
...
+	 *   Reserved (Bits [6:4])
+	 *
+	 *   Slot ID (Bits [3:0]):
+	 *       0001: Slot 1
+	 *       0010: Slot 2
+	 *       ...
+	 *       1000: Slot 8
+	 */
+
+	msg->data[0] = 0x80 + slot_id;
```
Comment (jagpalgill) [inline meta-facebook/yv4-sd/src/ipmi/plat_ipmi.c:146]: Whats 0x80? Can you please create a macro defining the value and then use it?

**Review by jagpalgill — CHANGES_REQUESTED** [review]

File: meta-facebook/yv4-sd/src/ipmi/plat_ipmi.c (hunk 1 above)
Comment (JeffShih-wiwynn) [reply meta-facebook/yv4-sd/src/ipmi/plat_ipmi.c]: We defined the enum SLOT_STATUS which contains PRESENT(0x80) and NOT_PRESENT(0x00).
Please check. Thanks.

Comment (jagpalgill) [reply meta-facebook/yv4-sd/src/ipmi/plat_ipmi.c]: Sure, please use these enums on line 146 then, rather than using static 0x80 then.

Comment (JeffShih-wiwynn) [reply meta-facebook/yv4-sd/src/ipmi/plat_ipmi.c]: We also change the part.
msg->data[0] = 0x80 + slot_id;
=>
msg->data[0] = SLOT_PRESENT + slot_id;

Please check.
Thanks.

**Review by jagpalgill — CHANGES_REQUESTED** [review]
1. Can you please rebase the diff.
2. I also notice a github build failure her - https://github.com/facebook/OpenBIC/actions/runs/8091450948/job/22110550876/. Can you please check.

**Comment by JeffShih-wiwynn** [comment]
> 1. Can you please rebase the diff.
> 2. I also notice a github build failure her - https://github.com/facebook/OpenBIC/actions/runs/8091450948/job/22110550876/. Can you please check.


1. We rebased it. Please check it. Thanks.
2. After confirming, it appears that the build failure for yv35-bb is not caused by this code. Please Check it. Thanks.

**Pull request closed without merging**